osu.Game/Screens/Ranking/ReplayDownloadButton.cs

```diff
@@ -87,19 +87,20 @@ private void load(OsuGame game, ScoreModelDownloader scores)
                     });
                 }
 
-                button.Enabled.Value = replayAvailability != ReplayAvailability.NotAvailable;
-                updateTooltip();
+                updateState();
             }, true);
 
             State.BindValueChanged(state =>
             {
                 button.State.Value = state.NewValue;
-                updateTooltip();
+                updateState();
             }, true);
         }
 
-        private void updateTooltip()
+        private void updateState()
         {
+            button.Enabled.Value = replayAvailability != ReplayAvailability.NotAvailable;
```

Comment:
I think it'd be better to add a `button.Enabled.Value` line in each of the `case` statements below, to ensure that every case is covered. Writing it this way makes it very hard to understand, since the code following it doesn't use the `NotAvailable` state directly